File: mobile/ios/guidr/CrashLogModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(CrashLogModule, NSObject)
RCT_EXTERN_METHOD(getEntries:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(clear:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
@end

File: mobile/ios/guidr/CrashLogModule.swift
import Foundation
import React

@objc(CrashLogModule)
class CrashLogModule: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool { return false }

  @objc func getEntries(
    _ resolve: @escaping RCTPromiseResolveBlock,
    rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    resolve(CrashLogStorage.shared.loadEntries())
  }

  @objc func clear(
    _ resolve: @escaping RCTPromiseResolveBlock,
    rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    CrashLogStorage.shared.clear()
    resolve(nil)
  }
}
